src/JobConsumer/Program.cs

```diff
@@ -0,0 +1,316 @@
+
+                        await Task.Delay(1000);
+                        continue;
+                    }
+
+                    Console.WriteLine($"Found '{nextFile.Name}'");
+
+                    // Attempting to move the file to the processing folder in order to lock it
+                    var session = nextFile.Name.Split('.', 2)[0];
+                    var processingFilePath = Path.Combine(ProcessingPath, nextFile.Name);
+                    var processingFile = new FileInfo(processingFilePath);
+
+                    // If we can't move the file to the processing folder, we continue, which might retry the same file
+                    try
+                    {
+                        nextFile.MoveTo(processingFilePath);
+                    }
+                    catch
+                    {
+                        Console.WriteLine($"The file named '{nextFile.FullName}' couldn't be moved to '{processingFilePath}'. Skipping...");
+                        continue;
+                    }
+
+                    var benchmarkResult = await BenchmarkPR(processingFile, session);
+                    await PublishResult(processingFile, benchmarkResult);
+                }
+            });
+
+            return app.Execute(args);
+        }
+
+        private static async Task<BenchmarkResult> BenchmarkPR(FileInfo processingFile, string session)
+        {
+            Directory.SetCurrentDirectory(RepoPath);
+
+            var outputBuilder = new StringBuilder();
+            var errorBuilder = new StringBuilder();
+
+            var buildRules = await GetBuildInstructions(processingFile);
+
+            Process.Start("git", "clean -xdf").WaitForExit();
+            Process.Start("git", "fetch").WaitForExit();
+            Process.Start("git", $"checkout {buildRules.BaselineSHA}").WaitForExit();
+
+            RunBuildCommands(buildRules);
+
+            var baselineArguments = $"{DriverPath} --server {ServerUrl} --client {ClientUrl} --jobs {processingFile.FullName} --session {session} --self-contained --save baseline --description Before";
+
+            outputBuilder.AppendLine($"Starting baseline run on '{buildRules.BaselineSHA}'...");
+            var baselineSuccess = await RunDriver(baselineArguments, outputBuilder, errorBuilder);
+
+            if (!baselineSuccess)
+            {
+                errorBuilder.AppendLine($"Baseline benchmark run on '{buildRules.BaselineSHA}' failed.");
+                return new BenchmarkResult
+                {
+                    Success = false,
+                    BaselineStdout = outputBuilder.ToString(),
+                    BaselineStderr = errorBuilder.ToString(),
+                };
+            }
+
+            var baselineStdout = outputBuilder.ToString();
+            var baselinseStderr = errorBuilder.ToString();
+            outputBuilder.Clear();
+            errorBuilder.Clear();
+
+            Process.Start("git", $"checkout {buildRules.PullRequestSHA}").WaitForExit();
```

Comment:
nit: GitHub merges the base branch and the PR branch when building, we might want to do something similar.